### PR TITLE
New version: Canonical.Ubuntu.2404 version 2404.0.5.0

### PR DESCRIPTION
**Pull request by jibel**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/158937)

### CODE REVIEW

**Comment by jibel** [comment]
I don't know why this URL is considered invalid.
This is the address of the package on Microsoft CDN from https://github.com/microsoft/WSL/blob/master/distributions/DistributionInfo.json and local validation of the manifest was successful.

**Comment by stephengillie** [comment]
Hi @jibel,

Thanks for sharing where you found the `InstallerUrl`.

I was trying to manually verify this last week. Sorry for not asking where the URL was found. We have a "navigation requirement" policy where the `InstallerUrl` can be found on a webpage by starting at the `PackageUrl`. 

In this case, I think waiving this is the best course. Using this location as a `PackageUrl` wouldn't be very informative, and would probably be a worse user experience.

File: manifests/c/Canonical/Ubuntu/2404/2404.0.5.0/Canonical.Ubuntu.2404.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+PackageIdentifier: Canonical.Ubuntu.2404
+PackageVersion: 2404.0.5.0
+MinimumOSVersion: 10.0.16215.0
+InstallerLocale: en-US
+InstallerType: appx
+Scope: user
+UpgradeBehavior: uninstallPrevious
+PackageFamilyName: CanonicalGroupLimited.Ubuntu24.04LTS_79rhkp1fndgsc
+Installers:
```
Comment (stephengillie) [inline manifests/c/Canonical/Ubuntu/2404/2404.0.5.0/Canonical.Ubuntu.2404.installer.yaml:12]:
```suggestion
Dependencies:
  WindowsFeatures:
    - Microsoft-Windows-Subsystem-Linux 
Installers:
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
It seems WSL might be needed:
image
image

(A list of Windows Feature names can be found with `Get-WindowsOptionalFeature -Online`.)

File: manifests/c/Canonical/Ubuntu/2404/2404.0.5.0/Canonical.Ubuntu.2404.installer.yaml (hunk 1 above)
Comment (jibel) [reply manifests/c/Canonical/Ubuntu/2404/2404.0.5.0/Canonical.Ubuntu.2404.installer.yaml:15]: Thanks for your review and update.

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 12:45:32 PM: 
> 2024-07-15 12:45:31.339 [CORE] Failed to enable Windows Feature [Microsoft-Windows-Subsystem-Linux] with exit code: 14081

Automated error analysis suggests 14081 may mean The referenced assembly could not be found. Is a dependency missing?

(Automated response - build 886.)

**Pull request closed without merging**